homeassistant/components/enocean/sensor.py

```diff
@@ -281,3 +292,22 @@ def value_changed(self, packet):
             self._attr_native_value = "tilt"
 
         self.schedule_update_ha_state()
+
+
+class EnOceanOpening(EnOceanSensor):
+    """Representation of an EnOcean opening sensor
+
+    EEPs (EnOcean Equipment Profiles):
+    - D5 00 01  (Contact sensor type NODON SDO - 2- 1 
+    """
+
+    def value_changed(self, packet):
+        """Update the internal state of the sensor."""
+        action = (packet.data[1] & 0x01)  # retrieve Contact (CO) value
```

Comment:
This is protocol details that should be extracted to a third party library. This integration has existing code that also violates this rule. Further changes to the existing code or new code that adds protocol details won't be accepted.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices

Reply:
OK, I'll do that when integrating these new features - thanks for your comment